File: app/src/main/java/com/iha/olmega_mobilesoftware_v2/AFEx/AcousticFeatureExtraction/StagePreHighpass.java
package com.iha.olmega_mobilesoftware_v2.AFEx.AcousticFeatureExtraction;

import android.util.Log;

import com.iha.olmega_mobilesoftware_v2.AFEx.Processing.Preprocessing.FilterHP;

import java.util.HashMap;

/**
 * Audio preprocessing
 */

public class StagePreHighpass extends Stage {

    final static String LOG = "StagePreHighpass";

    int cutoff_hz;

    FilterHP[] filterHP;


    public StagePreHighpass(HashMap parameter) {
        super(parameter);

        cutoff_hz = Integer.parseInt((String) parameter.get("cutoff_hz"));

        filterHP = new FilterHP[channels];
        for (int i = 0; i < channels; i++) {
            filterHP[i] = new FilterHP(samplingrate, cutoff_hz);
        }
    }

    @Override
    protected void process(float[][] buffer) {

        float[][] dataOut = new float[buffer.length][buffer[0].length];

        for (int i = 0; i < 2; i++) {
            System.arraycopy(buffer[i], 0, dataOut[i], 0, buffer[i].length);
            filterHP[i].filter(dataOut[i]);
        }

        send(dataOut);
    }

}

File: app/src/main/java/com/iha/olmega_mobilesoftware_v2/AFEx/AcousticFeatureExtraction/StagePreHighpass.kt
package com.iha.olmega_mobilesoftware_v2.AFEx.AcousticFeatureExtraction

import com.iha.olmega_mobilesoftware_v2.AFEx.Processing.Preprocessing.FilterHP

/**
 * Audio preprocessing
 */
class StagePreHighpass(parameter: HashMap<*, *>?) : Stage(parameter!!) {
    var cutoff_hz: Int
    var filterHP: Array<FilterHP?>?
    override fun process(buffer: Array<FloatArray>?) {
        val dataOut = Array<FloatArray>(buffer!!.size) { FloatArray(buffer[0].size) }
        for (i in 0..1) {
            System.arraycopy(buffer[i], 0, dataOut[i], 0, buffer[i].size)
            filterHP!![i]!!.filter(dataOut[i])
        }
        send(dataOut)
    }

    companion object {
        val LOG: String? = "StagePreHighpass"
    }

    init {
        cutoff_hz = parameter!!["cutoff_hz"].toString().toInt()
        filterHP = arrayOfNulls<FilterHP?>(Stage.Companion.channels)
        for (i in 0 until Stage.Companion.channels) {
            filterHP!![i] = FilterHP(Stage.Companion.samplingrate, cutoff_hz)
        }
    }
}